### PR TITLE
fix(ast): fix incorrect display for ast (part 1)

### PR DESCRIPTION
**Pull request by andylokandy**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

_Briefly describe what this PR aims to solve. Include background context that will help reviewers understand the purpose of the PR._

Related https://github.com/datafuselabs/databend/issues/14904

## Tests

- [x] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [x] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

### CODE REVIEW

**Comment by SkyFan2002** [comment]
May I ask how can we determine if the SQL generated by Display is equivalent to the original SQL? This has troubled me for a long time.

**Comment by andylokandy** [comment]
An ideal display will satisfy two rules: 1. The display string contains all information in the original sql, despite of formatting difference; 2. Reparse the displayed string and then display it again should come out identical sql.

In this PR, I'll add a check for the second rule.

**Comment by SkyFan2002** [comment]
> An ideal display will satisfy two rules: 1. The display string contains all information in the original sql, despite of formatting difference; 2. Reparse the displayed string and then display it again should come out identical sql.
> 
> In this PR, I'll add a check for the second rule.

Thanks. The second rule is brilliant.

File: src/query/ast/src/parser/parser.rs [hunk 1]
```diff
@@ -45,6 +45,34 @@ pub fn tokenize_sql(sql: &str) -> Result<Vec<Token>> {
...
+    //         Err(e) => {
+    //             let original_sql = tokens[0].source.to_string();
+    //             panic!(
+    //                 "Failed to reparse SQL:\n{}\nAST:\n{:#?}\n{}",
+    //                 original_sql, stmt.stmt, e
+    //             );
+    //         }
+    //     }
+    // }
+
```
Comment (leiysky) [inline src/query/ast/src/parser/parser.rs:75]: Remove these?

**Review by leiysky — APPROVED** [review]

File: src/query/ast/src/parser/parser.rs (hunk 1 above)
Comment (andylokandy) [reply src/query/ast/src/parser/parser.rs:75]: I'll enable it in the next PR

**Pull request merged**